distributed/scheduler.py

```diff
@@ -5038,6 +5049,19 @@ def stimulus_task_finished(self, key, worker, stimulus_id, run_id, **kwargs):
                     "stimulus_id": stimulus_id,
                 }
             ]
+        elif ts.state == "erred":
```

Comment:
Isn't this already covered by the next paragraph `elif ts.run_id != run_id`?

Reply:
No, AFAIR, we only issue a new `run_id` when the task is sent to a worker _again_.

Reply:
I've added another test for this. I don't the way this handled context managers, if you have suggestions on how to simplify this, I'm all ears.